### ISSUE TITLE
Array should be CvMat or IplImage in function 'cvGetSize'

### ISSUE THREAD

**Issue by kundanSingh11** [bytedeco/javacv#1129]
Exception in thread "main" java.lang.RuntimeException: OpenCV(4.0.1) C:\projects\javacpp-presets\opencv\cppbuild\windows-x86_64\opencv-4.0.1\modules\core\src\array.cpp:1246: error: (-5:Bad argument) Array should be CvMat or IplImage in function 'cvGetSize'

**Comment by saudet** [comment]
Just do what it says, pass a CvMat or an IplImage.

**Comment by kundanSingh11** [comment]
I am do just that

**Comment by saudet** [comment]
Ok, that's fine, please show your code. I can't guess what you're doing or not!

**Comment by kundanSingh11** [comment]
Getting error on this line :
IplImage destImage = downScaleImage(src, percent);
IplImage grayImage = IplImage.create(cvGetSize(destImage), IPL_DEPTH_8U, 1);

**Comment by saudet** [comment]
And what's in `downScaleImage()`?

**Comment by saudet** [comment]
And what is `src`? This works just fine:
```java
IplImage img = IplImage.create(200, 200, IPL_DEPTH_8U, 3);
downScaleImage(img, 50);
```
As expected, it outputs:
```
dest height 100 dest witdth 100
```
I can't fix what isn't broken! You'll need to give me something to work with here.

**Comment by kundanSingh11** [comment]
I have found a other way and its working good instead of using cvGetSize() i am using destImage.asCvMat().cvSize() and its working great. BTW great library.

**Comment by 970265** [comment]
Hi kundanSingh11, 

I have the same issue of "Array should be CvMat or IplImage in function 'cvGetSize'."
Can you let me know your solution?
Appreciated your kind support.

Yuan-Fa, Lee

**Comment by jessyuan24** [comment]
I have the same issue

**Comment by saudet** [comment]
I'm not able to reproduce this issue with JavaCV 1.5.2 and OpenCV 4.1.2, so it looks like this is fixed.

**Comment by saudet** [comment]
If this is still happening for you guys though, please complain upstream at https://github.com/opencv/opencv/issues/6221.

**Comment by kundanSingh11** [comment]
@jessyuan24 , @970265 I am using it as 
IplImage grayImage = IplImage.create(image.asCvMat().cvSize(), IPL_DEPTH_8U, 1);

**Comment by saudet** [comment]
BTW, this has now been fixed in with OpenCV 4.2.0, see issue https://github.com/opencv/opencv/issues/15990
